src/pixbuf.rs

```diff
@@ -194,7 +194,15 @@ impl Pixbuf {
         })
     }
 
-    pub unsafe fn get_pixels(&self) -> &mut [u8] {
+    pub fn get_pixels(&self) -> &[u8] {
+        unsafe {
+            let mut len = 0;
+            let ptr = ffi::gdk_pixbuf_get_pixels_with_length(self.to_glib_none().0, &mut len);
+            slice::from_raw_parts(ptr, len as usize)
+        }
+    }
+
+    pub unsafe fn get_pixels_mut(&self) -> &mut [u8] {
```

Comment:
This is (and was) wrong. You can create multiple mutable references to the same data, which is not allowed (even in unsafe blocks). The only thing you can return here is either an immutable reference or a mutable raw pointer.

Reply:
Should it be fixed in this pull request or later? I don't know how it should be implemented. I know that cairo-rs uses a runtime check for this.

Reply:
Also what is currently possible is to get an immutable and a mutable reference to the same data. Same problem really.

We could implement runtime checks here, like with `RefCell`, but it would be tricky. You would have to remember inside the `GObject` (maybe in the qdata?), otherwise one could simply clone (or create a new wrapper from the same raw pointer of the pixbuf via `from_glib_none`) and then circumvent the runtime check.

How is it done in cairo, do you have a link?

Reply:
@sdroege https://github.com/gtk-rs/cairo/blob/master/src/image_surface.rs#L62

Reply:
Kind of ugly and will make it impossible to get mutable access to the data if anybody else has a reference. Even if that other place does nothing at all with the data.

But I guess that's fine for now.

Reply:
I was thinking about the same that @RazrFalcon proposed. Instead we could maybe just keep a `RefCell` in the type directly with this data inside it and just return it when this method is called?